### PR TITLE
Update sample app to manage Default/Dark/Light theme taking account `PreferSystemTheme`

### PR DESCRIPTION
**Pull request by huaxing-yuan**
Update sample app to manage Default/Dark/Light theme while taking account `FluentAvaloniaTheme.PreferSystemTheme`.
https://github.com/amwx/FluentAvalonia/issues/409

With this update:
- If user chooses either Light and Dark theme, the theme will be fixed regardless system theme changes.
- If user chooses Default, then app theme will be synchronized with system theme

### CODE REVIEW

**Comment by workgroupengineering** [comment]
I tested this PR with #411 and #420. If #417 is applied, theme switching does not work.

**Comment by amwx** [comment]
I don't think this is the correct solution to the problem. `ThemeVariant.Default` has a very specific meaning in Avalonia and using to denote the system theme I think is a bad practice.

I need to take a closer look at your actual issue (not sure I really understand what's going on tbh) and see if I can come up with a better way. It may be this is an obscure issue in FATheme not properly handling system stuff.

**Comment by huaxing-yuan** [comment]
Thanks for the reply.

To reproduce the issue, we can launch the sample application, set it to dark mode. Then go to Windows and set default app mode to light. We can see sample application changes its theme to light, regardless we have chosen dark.

In my own application, I haven't used `ThemeVariant.Default`, but an enum to store user preferred themes between "System", "Dark" and "Light" with a mechanism to handle user settings by taking account both `PreferSystemTheme` and `RequestedThemeVariant`. The idea was: if user chooses a specific theme, then we need to set `PreferSystemTheme` to `false` so that system changes won't affect application. Do you think it's a more proper solution ?

Anyway, I was just too lazy to add `ThemeVariant.Default` in the list without knowing its specific meaning.

**Comment by amwx** [comment]
Oh I see, so this is more an issue with the sample app rather than an actual issue. 
In that case, I think the best course of action here is to
- change the ViewModel to use `string`s instead of `ThemeVariant`, with the options "System", "Light", and "Dark"
- Change `CurrentAppTheme` to also be a string, and in the setter:
  - if `System` is set, set `PreferSystemTheme` to true, which should automatically invalidate the current theme to match the system
  - if `Light` or `Dark` is set, ensure `PreferSystemTheme` is false and set the application's theme variant.

I did notice some issues within FluentAvaloniaTheme though, including an event handler leak with this, so if you don't mind, I'll fix up FATheme and push those changes to this branch within the next couple days.

**Comment by amwx** [comment]
I've just merged in the fixes for FATheme (#434), if you'd like to make the suggested changes feel free to go ahead-you can either undo the current changes in the PR and keep it in this one, or you can close this and do a new PR, whichever you prefer.

**Comment by huaxing-yuan** [comment]
Thanks, I'll update this PR and test your fix. (sorry for the late reply, I'm in vocation mode recently)

**Comment by huaxing-yuan** [comment]
Updated as discussed.
Tested locally on my Windows Laptop, but unable to test on Linux and Mac from here :(

File: samples/FAControlsGallery/ViewModels/SettingsPageViewModel.cs [hunk 1]
```diff
@@ -10,30 +10,58 @@ namespace FAControlsGallery.ViewModels;
 
 public class SettingsPageViewModel : MainPageViewModelBase
 {
+    readonly FluentAvaloniaTheme faTheme;
```
Comment (amwx) [inline samples/FAControlsGallery/ViewModels/SettingsPageViewModel.cs:13]: minor: `private readonly FluentAvaloniaTheme _faTheme;`, and move it to the bottom with the other fields

**Review by amwx — APPROVED** [review]
Generally looks good, thanks! If it works on Windows, I see no reason why it wouldn't work on another OS.

File: samples/FAControlsGallery/ViewModels/SettingsPageViewModel.cs (hunk 1 above)
Comment (huaxing-yuan) [reply samples/FAControlsGallery/ViewModels/SettingsPageViewModel.cs]: ok, done.

**Comment by huaxing-yuan** [comment]
Done, I've tested the first commit on Windows, Ubuntu and MacOS, it should work on at least these 3 platforms.

**Pull request merged**